### PR TITLE
feat: implement row striping options

### PR DESCRIPTION
**Pull request by rich-iannone**
This PR addresses https://github.com/posit-dev/great-tables/issues/435 by making row striping options functional in rendered tables. An example of using row striping involves three new options in `tab_options()`:

```python
from great_tables import GT, exibble

(
    GT(exibble[["num", "char", "row", "group"]], rowname_col="row", groupname_col="group")
    .tab_options(
        row_striping_include_table_body=True,
        row_striping_include_stub=True,
        row_striping_background_color="lightblue"
    )
)
```

image

Alternatively, the `opt_row_striping()` method allows for a simplified shortcut for `row_striping_include_table_body=True` so the user doesn't have to look through `tab_options()` for this basic functionality:

```python
from great_tables import GT, exibble

(
    GT(exibble[["num", "char", "row", "group"]], rowname_col="row", groupname_col="group")
    .opt_row_striping()
)
```

image

Note that the default background color for striping is a *very* light gray (by design).

### CODE REVIEW

File: tests/__snapshots__/test_options.ambr [hunk 1]
```diff
@@ -701,3 +701,163 @@
   
   '''
 # ---
+# name: test_tab_options_striping_snap
```
Comment (machow) [inline tests/__snapshots__/test_options.ambr:704]: This is too big. Can you reduce it to a snapshot that is mostly limited to the piece being tested, or test it another way?

File: tests/test_options.py [hunk 2]
```diff
@@ -387,6 +387,36 @@ def test_opt_table_font_raises():
...
+    assert gt_tbl_1._options.row_striping_include_table_body.value == True
+    assert gt_tbl_2._options.row_striping_include_table_body.value == False
+
+
+def test_tab_options_striping():
+
+    gt_tbl_opt = GT(exibble, id="test").opt_row_striping()
+    gt_tbl_tab_options = GT(exibble, id="test").tab_options(row_striping_include_table_body=True)
+
+    assert gt_tbl_opt.as_raw_html() == gt_tbl_tab_options.as_raw_html()
```
Comment (machow) [inline tests/test_options.py:406]: Can you test this further upstream? If rendered html strings are the very last step, is there an earlier step that indicates these two approaches do the same thing (e.g. checking the options values directly, etc..)?

File: tests/test_options.py [hunk 3]
```diff
@@ -387,6 +387,36 @@ def test_opt_table_font_raises():
...
+
+
+def test_tab_options_striping_snap(snapshot):
```
Comment (machow) [inline tests/test_options.py:409]: See comment on snapshot. It'd be okay to test the css and html separately if it makes it easier to strip down. I wonder if you could easily test the css using an assertion, and snapshot test just the html body (with fewer rows of data; the minimum needed to test--3? or w/e makes sense given there are groupings)

**Review by machow — CHANGES_REQUESTED** [review]

File: tests/test_options.py (hunk 2 above)
Comment (rich-iannone) [reply tests/test_options.py:406]: I'll check the options values directly here (comparing the two approaches).

File: tests/test_options.py (hunk 3 above)
Comment (rich-iannone) [reply tests/test_options.py:409]: I'm going to make the input df with much smaller dimensions and only snapshot the table body (everything inside `<tbody>`). I don't think there's a need to snapshot the CSS part.

File: great_tables/_utils_render_html.py [hunk 4]
```diff
@@ -503,10 +515,30 @@ def create_body_component_h(data: GTData) -> str:
                 cell_styles = ""
 
             if is_stub_cell:
-                body_cells.append(f"""    <th class="gt_row gt_left gt_stub">{cell_str}</th>""")
+
```
Comment (machow) [inline great_tables/_utils_render_html.py:518]: This is looking good, so just a small nit here 😅. Do you mind consolidating so there is a bit less duplicated code? Seems like the following could happen:

* change variables like `td_classes` and `th_classes` to have the same name. Similar with `th_classes_str`, etc..
* add a variable like el_name that is either "th" or "td"
* move all other logic (e.g. the appending of "gt_striped", the " ".join, the body_cells.append) out of the if / else, so it is only written once (e.g. move it to line 543 and unindent)
  - Use el_name in what should now be a single piece of code creating the final th or td string.

It looks like if something is a stub, that only affects the element name and some of the classes to be set. (But other logic is being included)

File: tests/test_options.py [hunk 5]
```diff
@@ -387,6 +388,45 @@ def test_opt_table_font_raises():
...
+    assert gt_tbl_opt_stri._options.row_striping_include_stub.value == False
+
+
+def test_tab_options_striping_snap(snapshot):
+
+    gt_tbl = GT(
+        exibble[["row", "group", "char"]].head(4), rowname_col="row", groupname_col="group"
+    ).tab_options(
+        row_striping_include_table_body=True,
+        row_striping_include_stub=True,
```
Comment (machow) [inline tests/test_options.py:420]: Can you test these separately, so that we know if one options is specified, it does not set both things?

**Review by machow — CHANGES_REQUESTED** [review]

File: great_tables/_utils_render_html.py (hunk 4 above)
Comment (rich-iannone) [reply great_tables/_utils_render_html.py:518]: Thanks, will make these changes!

File: tests/test_options.py (hunk 5 above)
Comment (rich-iannone) [reply tests/test_options.py]: Good idea. There is now two distinct snapshots where one tests striping only of the stub, the other only of the body cells.

**Review by machow — APPROVED** [review]
Yo, thanks for all these tweaks, it's beautiful!

**Comment by rich-iannone** [comment]
Thanks @machow! Always a pleasure.

**Pull request merged**